File: questApplication/src/main/java/ru/nekit/android/qls/quest/mediator/types/simpleExample/SimpleExampleQuestContentMediator.java
package ru.nekit.android.qls.quest.mediator.types.simpleExample;

import android.support.annotation.NonNull;
import android.view.View;
import android.widget.EditText;

import ru.nekit.android.qls.quest.QuestContext;
import ru.nekit.android.qls.quest.formatter.IQuestTextContentFormatter;
import ru.nekit.android.qls.quest.formatter.TextQuestContentFormatter;
import ru.nekit.android.qls.quest.mediator.content.AbstractQuestContentMediator;

import static android.view.View.GONE;

public class SimpleExampleQuestContentMediator extends AbstractQuestContentMediator {

    protected SimpleExampleQuestViewHolder mViewHolder;

    @NonNull
    protected IQuestTextContentFormatter createFormatter() {
        return new TextQuestContentFormatter();
    }

    @Override
    public void onCreate(@NonNull QuestContext questContext) {
        super.onCreate(questContext);
        mViewHolder = new SimpleExampleQuestViewHolder(mQuestContext);
        IQuestTextContentFormatter formatter = createFormatter();
        String[] questStringList = formatter.format(mQuestContext, mQuest);
        switch (mQuest.getQuestionType()) {

            case SOLUTION:

                mViewHolder.rightSideView.setVisibility(GONE);
                mViewHolder.leftSideView.setText(questStringList[0]);
                mViewHolder.alternativeAnswerInput.setHint(formatter.getMissedCharacter());

                break;

            case UNKNOWN_MEMBER:

                mViewHolder.leftSideView.setText(questStringList[0]);
                mViewHolder.rightSideView.setText(questStringList[1]);
                mViewHolder.alternativeAnswerInput.setHint(formatter.getMissedCharacter());

                break;

            case COMPARISON:

                mViewHolder.alternativeAnswerInput.setVisibility(GONE);
                mViewHolder.rightSideView.setVisibility(GONE);
                mViewHolder.leftSideView.setText(questStringList[0]);

                break;

            case UNKNOWN_OPERATION:

                mViewHolder.alternativeAnswerInput.setVisibility(GONE);
                mViewHolder.rightSideView.setVisibility(GONE);
                mViewHolder.leftSideView.setText(questStringList[0]);

                break;

        }
    }

    @Override
    public void onQuestPlay(boolean delayedPlay) {
        super.onQuestPlay(delayedPlay);
    }

    @Override
    public View getView() {
        return mViewHolder.getView();
    }

    @Override
    public EditText getAnswerInput() {
        return mViewHolder.alternativeAnswerInput;
    }

    @Override
    public boolean includeInLayout() {
        return true;
    }

    @Override
    public void updateSize() {

    }

}

File: questPresentation/src/main/java/ru/nekit/android/qls/quest/view/mediator/types/simpleExample/SimpleExampleQuestContentMediator.kt
package ru.nekit.android.qls.quest.view.mediator.types.simpleExample

import android.view.View
import android.view.View.GONE
import android.widget.EditText
import ru.nekit.android.qls.domain.model.quest.Quest
import ru.nekit.android.qls.quest.QuestContext
import ru.nekit.android.qls.quest.formatter.IQuestTextContentFormatter
import ru.nekit.android.qls.quest.formatter.TextQuestContentFormatter
import ru.nekit.android.qls.quest.types.TextQuest
import ru.nekit.android.qls.quest.view.mediator.content.SimpleContentMediator
import ru.nekit.android.qls.shared.model.QuestionType

//ver 1.0
open class SimpleExampleQuestContentMediator : SimpleContentMediator() {

    private lateinit var _quest: TextQuest
    protected lateinit var viewHolder: SimpleExampleQuestViewHolder

    override val view: View?
        get() = viewHolder.view

    override val answerInput: EditText?
        get() = viewHolder.alternativeAnswerInput

    protected open fun createFormatter(): IQuestTextContentFormatter {
        return TextQuestContentFormatter()
    }

    override fun onCreate(questContext: QuestContext, quest: Quest) {
        super.onCreate(questContext, quest)
        viewHolder = SimpleExampleQuestViewHolder(questContext)
        val formatter = createFormatter()
        _quest = quest as TextQuest
        with(_quest) {
            val questStringList = formatter.format(questContext, this)
            when (questionType) {

                QuestionType.SOLUTION -> {

                    viewHolder.rightSideView.visibility = GONE
                    viewHolder.leftSideView.text = questStringList[0]
                    viewHolder.alternativeAnswerInput.hint = formatter.missedCharacter
                }

                QuestionType.UNKNOWN_MEMBER -> {

                    viewHolder.rightSideView.text = questStringList[0]
                    viewHolder.leftSideView.text = questStringList[1]
                    viewHolder.alternativeAnswerInput.hint = formatter.missedCharacter
                }

                QuestionType.COMPARISON -> {

                    viewHolder.alternativeAnswerInput.visibility = GONE
                    viewHolder.rightSideView.visibility = GONE
                    viewHolder.leftSideView.text = questStringList[0]
                }

                QuestionType.UNKNOWN_OPERATION -> {

                    viewHolder.alternativeAnswerInput.visibility = GONE
                    viewHolder.rightSideView.visibility = GONE
                    viewHolder.leftSideView.text = questStringList[0]
                }
            }
        }
    }


    override fun includeInLayout(): Boolean {
        return true
    }

    override fun updateSize() {

    }

}
